Conference-Android/app/src/main/java/com/sagetech/conference_android/app/ui/presenter/ConferenceSessionListActivityPresenter.java

```diff
@@ -41,13 +46,43 @@ public void onError(Throwable e) {
             }
 
             @Override
-            public void onNext(List<ConferenceSessionData> conferenceSessionDatas) {
-                conferenceSessionListActivity.populateConferenceSessions(ConferenceSessionViewBuilder.toConferenceSessionViewModel(conferenceSessionDatas));
+            public void onNext(List<ConferenceSessionViewModel> conferenceSessionViewModels) {
+                conferenceSessionListActivity.populateConferenceSessions(conferenceSessionViewModels);
             }
 
         });
     }
 
+    private Observable<List<ConferenceSessionViewModel>> createConferenceSessionViewModelObservable(Integer conferenceId) {
+
+        Observable<List<ConferenceSessionData>> conferenceSessionObservable = conferenceController.getConferenceSessionsById(conferenceId).cache();
+
+        Observable<Map<Long, RoomData>> roomDataObservable = conferenceSessionObservable.flatMap(new Func1<List<ConferenceSessionData>, Observable<ConferenceSessionData>>() {
+            @Override
+            public Observable<ConferenceSessionData> call(List<ConferenceSessionData> conferenceSessionDatas) {
+                return Observable.from(conferenceSessionDatas);
+            }
+        }).flatMap(new Func1<ConferenceSessionData, Observable<RoomData>>() {
+            @Override
+            public Observable<RoomData> call(ConferenceSessionData conferenceSessionData) {
+                return conferenceController.getRoomById(conferenceSessionData.getRoomId());
+            }
+        }).toMap(new Func1<RoomData, Long>() {
```

Comment:
somehow in this, if we were going to be extremely efficient, we should probably do a _distinct_ call in order to make sure that we are not calling the getRoomById more than we need to.

http://reactivex.io/documentation/operators/distinct.html